src/components/filter-panel/categories/categories.tsx

```diff
@@ -26,18 +28,32 @@ export function CategoriesNavigation() {
   } = useCategoryNavigationStore();
 
   useEffect(() => {
-    const fetchCategories = async (): Promise<void> => {
-      const categories: Category[] = await getCategories();
-      if (categories) {
-        setAllCategories(categories);
-      } else {
-        setReset();
-      }
-    };
-    void fetchCategories();
-  }, [setReset, setAllCategories]);
-
-  const handleRootCategoryClick = (name: string): void => {
+    getCategories()
+      .then((categories) => {
+        if (categories) {
+          setAllCategories(categories);
+        } else {
+          setReset();
+        }
+      })
+      .catch((e: Error) => console.error("can't get categories", e))
+      .finally(() => {
+        if (redirectInToMainPage) {
+          handleCategoryRedirect(redirectInToMainPage);
+        }
+      });
+  }, [setReset, setAllCategories, redirectInToMainPage]);
+
+  function handleCategoryRedirect(redirectInToMainPage: string) {
+    if (
+      (redirectInToMainPage === 'Man' || redirectInToMainPage === 'Woman') &&
```

Comment:
As I see it from code there is only two values for redirectInToMainPage variable: 'Man' and 'Woman'.
```suggestion
      redirectInToMainPage !== '' &&
```